bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2342,21 +2341,20 @@ void ReplicaImp::onMessage<ViewChangeMsg>(ViewChangeMsg *msg) {
       }
     } else {
       if (complaintMsg->viewNumber() + 1 == msg->newView() && validateMessage(complaintMsg.get())) {
-        complainedReplicasForHigherView.store(std::move(complaintMsg));
+        viewsManager->getComplaintsForHigherView().store(std::move(complaintMsg));
       } else {
         LOG_WARN(VC_LOG, "Invalid complaint in ViewChangeMsg for a higher View.");
       }
     }
   }
-  if (complainedReplicasForHigherView.hasQuorumToLeaveView()) {
-    ConcordAssert(msg->newView() > getCurrentView() + 1);
-    complainedReplicas = std::move(complainedReplicasForHigherView);
-    LOG_INFO(
-        VC_LOG,
-        "Got quorum of Replicas complaining for a higher View in VCMsg: " << KVLOG(msg->newView(), getCurrentView()));
+
+  if (viewsManager->shouldJumpToHigherViewBasedOnQuorumOfComplaints(msg)) {
     MoveToHigherView(msg->newView());
   }
 
+  // The container has to be empty for the next View change message.
+  viewsManager->getComplaintsForHigherView().clear();
```

Comment:
can we have a separate `cleanComplaintsForHigherView` method, to avoid calling `.clear()` on a private member?